src/hotspot/share/oops/compressedKlass.cpp

```diff
@@ -32,9 +32,31 @@
 address CompressedKlassPointers::_base = nullptr;
 int CompressedKlassPointers::_shift = 0;
 size_t CompressedKlassPointers::_range = 0;
+// Note: initialization value is unchanged for -UseCompressedClassPointers, so
+// the bit mirroring UseCompressedClassPointers is off and maches the switch.
+uint64_t CompressedKlassPointers::_compressionInfo;
 
 #ifdef _LP64
 
+void CompressedKlassPointers::set_base_and_shift(address thebase, int theshift) {
+  assert(UseCompressedClassPointers, "only for compressed klass code");
+  assert(theshift == 0 || theshift == LogKlassAlignmentInBytes, "invalid shift for klass ptrs");
+  _base = thebase;
+  _shift = theshift;
```

Comment:
Do we even need `_base` and `_shift` as separate fields after this change then?

Reply:
No, I don't think so. Lets remove it.

Reply:
Wait, yes, we still need it for VMStructs. Would require duplicating the decoding logic on the VMStructs consumer side, which I would rather not.

Reply:
Oh, okay.